From dd5adbf2093c69efd4137b0095b908d46ffdac32 Mon Sep 17 00:00:00 2001
From: Rahul Bhattacharjee <quic_rbhattac@quicinc.com>
Date: Fri, 8 Jul 2022 18:49:52 +0530
Subject: [PATCH] ath11k: fix 6g station association failure

In 6G STA, setting PEER_AUTH flag before completion of 4 way handshake causing
fw to drop M4 packet(EAPOL) and it results in association failure.

In general, PEER_AUTH flag should not to be set before completion of 4 way
handshake for security mode and clear the PEER_AUTH flag based on rsnie
presence to know security or open mode.

For 6G, clearing the PEER_AUTH bit irrespective of rsnie presense because it
does not support open mode.

Signed-off-by: Thiraviyam Mariyappan <quic_tmariyap@quicinc.com>
Signed-off-by: Rahul Bhattacharjee <quic_rbhattac@quicinc.com>
---
 drivers/net/wireless/ath/ath11k/mac.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/wireless/ath/ath11k/mac.c b/drivers/net/wireless/ath/ath11k/mac.c
index 85fe155..4c58281 100644
--- a/drivers/net/wireless/ath/ath11k/mac.c
+++ b/drivers/net/wireless/ath/ath11k/mac.c
@@ -895,7 +895,7 @@ static void ath11k_peer_assoc_h_crypto(struct ath11k *ar,
 	}
 
 	/* FIXME: base on RSN IE/WPA IE is a correct idea? */
-	if (rsnie || wpaie) {
+	if (ar->supports_6ghz || rsnie || wpaie) {
 		ath11k_dbg(ar->ab, ATH11K_DBG_WMI,
 			   "%s: rsn ie found\n", __func__);
 		arg->need_ptk_4_way = true;
-- 
2.35.1

